### PR TITLE
various changes for data range compactions

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

* add database-global compact operation via REST endpoint PUT `/_admin/compact` and JavaScript API `db._compact();`.
* introduce compaction on active failover followers after they process a truncate command and the truncation affected 4k documents or more.

- [x] :hankey: Bugfix 
- [x] :pizza: New feature 
- [ ] :hammer: Refactoring 
- [x] :book: CHANGELOG entry made
- [x] :muscle: The behavior in this PR was *manually tested*
- [ ] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [ ] No backports required
- [ ] Backports required for: 3.6

#### Related Information

- [x] Docs PR: https://github.com/arangodb/docs/pull/537
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket number: https://arangodb.atlassian.net/browse/ES-657
- [ ] Design document: 

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11702/

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -4,6 +4,26 @@ v3.7.3 (XXXX-XX-XX)
...
+* Make followers in active failover run a compaction after they process a
+  truncate operation and the truncate removed more than 4k documents. This
+  can help to reclaim disk space on the follower earlier than without running
+  the truncate.
+
+* The REST API PUT `/_api/collection/<name>/truncate` will now also run a
+  compaction if the truncation affected more than 4k documents. This may add
+  extra latency to the truncate operation, but can help to reclaim disk space
+  earlier.
+
```
Comment (jsteemann) [inline CHANGELOG:16]:
```suggestion
```

**Comment by graetzer** [comment]
why is this not implement as a storage level API with exactly one rest handler

**Comment by graetzer** [comment]
and why not compact() automatically as we have done before

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11716/

**Review by graetzer — APPROVED** [review]

**Comment by jsteemann** [comment]
Tests blue except known failure on Linux

**Pull request merged**